src/pages/TokenDetails/Widget.tsx

```diff
@@ -0,0 +1,183 @@
+    return { slippage: { auto, max: slippage }, transactionTtl: ttl }
+  }, [slippage, ttl, userSlippage])
+  const settingsHandlers: SwapEventHandlers = useMemo(
+    () => ({ onSettingsReset, onSlippageChange, onTransactionDeadlineChange }),
+    [onSettingsReset, onSlippageChange, onTransactionDeadlineChange]
+  )
+
+  return { settings: { settings, ...settingsHandlers } }
+}
+
+// Integrates the Widget's controlled value, using the app's CurrencySearchModal for token selection.
+function useSyncWidgetValue(defaultToken: Currency) {
+  const [isExactInput, setIsExactInput] = useState(false)
+  const [amount, setAmount] = useState<string>()
+  const onAmountChange = useCallback((field: Field, amount: string) => {
+    setIsExactInput(field === Field.INPUT)
+    setAmount(amount)
+  }, [])
+  const [tokens, setTokens] = useState<{ [Field.INPUT]?: Currency; [Field.OUTPUT]?: Currency }>({
+    [Field.OUTPUT]: defaultToken,
+  })
+  const onSwitchTokens = useCallback(() => {
+    setIsExactInput((isExactInput) => !isExactInput)
+    setTokens((tokens) => ({
+      [Field.INPUT]: tokens[Field.OUTPUT],
+      [Field.OUTPUT]: tokens[Field.INPUT],
+    }))
+  }, [])
+  const [selectingField, setSelectingField] = useState<Field>()
+  const otherField = useMemo(() => (selectingField === Field.INPUT ? Field.OUTPUT : Field.INPUT), [selectingField])
+  const [selectingToken, otherToken] = useMemo(() => {
+    if (selectingField === undefined) return [undefined, undefined]
+    return [tokens[selectingField], tokens[otherField]]
+  }, [otherField, selectingField, tokens])
+  const onTokenSelectorClick = useCallback((field: Field) => {
+    setSelectingField(field)
+    return false
+  }, [])
+  const onTokenSelect = useCallback(
+    (token: Currency) => {
+      if (selectingField === undefined) return
+      setIsExactInput(selectingField === Field.INPUT)
+      setTokens(() => {
+        return {
+          [otherField]: token === otherToken ? selectingToken : otherToken,
+          [selectingField]: token,
+        }
+      })
+    },
+    [otherField, otherToken, selectingField, selectingToken]
+  )
+  const value: SwapController = useMemo(
+    () => ({ type: isExactInput ? TradeType.EXACT_INPUT : TradeType.EXACT_OUTPUT, amount, ...tokens }),
+    [amount, isExactInput, tokens]
+  )
+  const valueHandlers: SwapEventHandlers = useMemo(
+    () => ({ onAmountChange, onSwitchTokens, onTokenSelectorClick }),
+    [onAmountChange, onSwitchTokens, onTokenSelectorClick]
+  )
+
+  const tokenSelector = (
+    <CurrencySearchModal
+      isOpen={selectingField !== undefined}
+      onDismiss={() => setSelectingField(undefined)}
+      selectedCurrency={selectingToken}
+      otherSelectedCurrency={otherToken}
+      onCurrencySelect={onTokenSelect}
+    />
+  )
+  return { value: { value, ...valueHandlers }, tokenSelector }
+}
+
+// Integrates the Widget's transactions, showing the widget's transactions in the app.
+function useSyncWidgetTransactions() {
```

Comment:
aren't these constant? not sure why they need to be memo'd or even a hook

Reply:
This is just a placeholder for a future integration that @JFrankfurt is working on